pkg/otelcgen/otelcgen.go

```diff
@@ -0,0 +1,202 @@
+package otelcgen
+
+import (
+	"fmt"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/confmap"
+	"go.opentelemetry.io/collector/otelcol"
+	"gopkg.in/yaml.v3"
+)
+
+// HistogramConfig is based on:
+// "go.opentelemetry.io/opentelemetry-collector-contrib/receiver/statsdreceiver/internal/protocol/statsd_parser.go.HistogramConfig"
+// with reduced number of attributes to reduce the number of dependencies.
+type HistogramConfig struct {
+	MaxSize int32 `mapstructure:"max_size"`
+}
+
+// TimerHistogramMapping is based on:
+// "go.opentelemetry.io/opentelemetry-collector-contrib/receiver/statsdreceiver/internal/protocol/statsd_parser.go.TLSSetting"
+// with reduced number of attributes to reduce the number of dependencies.
+type TimerHistogramMapping struct {
+	StatsDType   string          `mapstructure:"statsd_type"`
+	ObserverType string          `mapstructure:"observer_type"`
+	Histogram    HistogramConfig `mapstructure:"histogram"`
+}
+
+// TLSSetting is based on:
+// "go.opentelemetry.io/collector/config/configtls.TLSSetting"
+// with reduced number of attributes to reduce the number of dependencies.
+type TLSSetting struct {
+	CAFile   string `mapstructure:"ca_file,omitempty"`
+	KeyFile  string `mapstructure:"key_file,omitempty"`
+	CertFile string `mapstructure:"cert_file,omitempty"`
+}
+
+// ServerConfig is based on "go.opentelemetry.io/collector/config/confighttp.ServerConfig" and
+// "go.opentelemetry.io/collector/config/confighttp.ServerConfig" with reduced number of attributes
+// to reduce the number of dependencies.
+type ServerConfig struct {
+	// TLSSetting struct exposes TLS client configuration.
+	TLSSetting *TLSSetting `mapstructure:"tls,omitempty"`
+
+	// Additional headers attached to each HTTP request sent by the client.
+	// Existing header values are overwritten if collision happens.
+	// Header values are opaque since they may be sensitive.
+	Headers map[string]string `mapstructure:"headers,omitempty"`
+
+	// The target URL to send data to (e.g.: http://some.url:9411/v1/traces).
+	Endpoint string `mapstructure:"endpoint"`
+}
+
+type Protocol string
+
+type Protocols []Protocol
+
+const (
+	JaegerProtocol Protocol = "jaeger"
+	ZipkinProtocol Protocol = "zipkin"
+	OtlpProtocol   Protocol = "otlp"
+	StatsdProtocol Protocol = "statsd"
+)
+
+var (
+	JaegerID = component.MustNewID(string(JaegerProtocol))
+	OtlpID   = component.MustNewID(string(OtlpProtocol))
+	StatsdID = component.MustNewID(string(StatsdProtocol))
+	ZipkinID = component.MustNewID(string(ZipkinProtocol))
+)
+
+type Config struct {
+	cfg     *otelcol.Config
+	tlsKey  string
+	tlsCert string
+	podIP   string
+}
+
+type Option func(c *Config) error
+
+func NewConfig(options ...Option) (*Config, error) {
```

Comment:
are there mandatory values? if so, please pass them in explicitly

Reply:
fixed.